tests/test_netcdf_dataset.py

```diff
@@ -77,3 +77,91 @@ def test_get_dataloaders_aws():
         data = next(t)
 
         assert "sat_data" in data.keys()
+
+
+@pytest.mark.skip("CD does not have access to GCP")
+def test_required_keys_gcp():
+
+    DATA_PATH = "gs://solar-pv-nowcasting-data/prepared_ML_training_data/v5/"
+    TEMP_PATH = "../nowcasting_dataset"
+    if os.path.isdir(os.path.join(TEMP_PATH, "train")):
+        os.removedirs(os.path.join(TEMP_PATH, "train"))
+    os.mkdir(os.path.join(TEMP_PATH, "train"))
+
+    train_dataset = NetCDFDataset(
+        24_900,
+        os.path.join(DATA_PATH, "train"),
+        os.path.join(TEMP_PATH, "train"),
+        cloud="gcp",
+        required_keys=[
+            "nwp",
+            "nwp_x_coords",
+            "nwp_y_coords",
+            "sat_data",
+            "sat_x_coords",
+            "sat_y_coords",
+        ],
+    )
+
+    dataloader_config = dict(
+        pin_memory=True,
+        num_workers=2,
+        prefetch_factor=8,
+        worker_init_fn=worker_init_fn,
+        persistent_workers=True,
+        # Disable automatic batching because dataset
+        # returns complete batches.
+        batch_size=None,
+    )
+
+    _ = torch.utils.data.DataLoader(train_dataset, **dataloader_config)
+
+    train_dataset.per_worker_init(1)
+    t = iter(train_dataset)
+    data = next(t)
+
+    assert "sat_data" in data.keys()
+    assert "pv_yield" not in data.keys()
+
+
+@pytest.mark.skip("CD does not have access to GCP")
```

Comment:
Good test - at some point we'll have to give it access to GCP. or maybe we download some .nc data and use this to run unit tests. Maybe we could make that an issue for now

Reply:
We could try with some actions like these maybe? https://github.com/google-github-actions/setup-gcloud but yeah, I might add a small .nc of satellite data for unittesting the subsetting.